### PR TITLE
Internal: GitHub - Publish Cloud Version [ED-7614] 

### PR DESCRIPTION
**Pull request by maork-elementor**

### CODE REVIEW

File: .github/workflows/publish-cloud.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+          prerelease: true
+          body_path: temp-changelog.txt
+        env:
+          GITHUB_TOKEN: ${{ secrets.MAINTAIN_TOKEN }}
+      - name: Send slack message
+        uses: ./.github/workflows/slack-notify
+        with:
+          MESSAGE: "@here Elementor ${{ env.PACKAGE_VERSION }} has been released! Saddle up partners! :tada:"
+          SLACK_TOKEN: ${{ secrets.SLACK_TOKEN }}
+          SLACK_CHANNEL: ${{ 'C03PMU9HK24' }}
```
Comment (matipojo) [inline .github/workflows/publish-cloud.yml:105]: Use from secrets
```suggestion
          SLACK_CHANNEL: ${{ 'C03PMU9HK24' }}
```

File: .github/workflows/push-new-version/action.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Push new version
+description: Commit and push the new version to the remote repository after the "Bump version" workflow has been completed.
```
Comment (matipojo) [inline .github/workflows/push-new-version/action.yml:2]: Please make it generic or find an existing action to make a git commit.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-cloud.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
+name: Publish Cloud Version
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/publish-cloud.yml:1]: But all the other yml files are not contains "version"

Screen Shot 2022-07-21 at 11 10 47

I will change the name itself.

File: .github/workflows/build-plugin/action.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - run: |
+            if [[ -z "$PACKAGE_VERSION" ]]; then
+              echo "Missing PACKAGE_VERSION env var"
+              exit 1
+            fi
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/build-plugin/action.yml:15]: done

File: .github/workflows/build-plugin/action.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+name: Build plugin
+description: Clean readme file and filter out unwanted content, The plugin will be built and uploaded to the Github artifacts directory.
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/build-plugin/action.yml:2]: done

File: .github/workflows/config.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "envirement": "production", 
+  "deployment_permitted" : "ronkelementor,KingYes,shilo-ey,matipojo,",
+  "repository" : "elementor"
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/config.json:4]: done

File: .github/workflows/create-tag/action.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+name: Create tag
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/create-tag/action.yml:1]: done

File: .github/workflows/create-tag/action.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+description: Generate a new tag and push it to the remote repository.
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/create-tag/action.yml:2]: done

File: .github/workflows/publish-cloud.yml [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        uses: antifree/json-to-variables@v1.0.1
+        with: 
+            filename: './.github/workflows/config.json'
+            prefix: config
+      - name: Test permissions
+        uses: ./.github/workflows/permissions
+      - name: Install Node.js 14.x
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14.x
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/publish-cloud.yml:29]: done

File: .github/workflows/publish-cloud.yml [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        uses: ./.github/workflows/build-plugin
+        with:
+          PACKAGE_VERSION: ${{ env.PACKAGE_VERSION }}
+      - name: Upload zip file to GitHub actions artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: elementor-${{ env.PACKAGE_VERSION }}
+          path: ${{ github.workspace }}/elementor/**/*
+          if-no-files-found: error
+      - name: Create Git tag
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/publish-cloud.yml:78]: done

File: .github/workflows/publish-cloud.yml (hunk 1 above)
Comment (maork-elementor) [reply .github/workflows/publish-cloud.yml:105]: done

File: .github/workflows/push-new-version/action.yml (hunk 2 above)
Comment (maork-elementor) [reply .github/workflows/push-new-version/action.yml:2]: done

File: .github/workflows/build-plugin/action.yml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+            sed -i -E "s/Stable tag: .*/Stable tag: ${PACKAGE_VERSION}/g" ./readme.txt
+            sed -i -E "s/Version: .*/Version: ${PACKAGE_VERSION}/g" elementor.php
+            sed -i -E "s/ELEMENTOR_VERSION', '.*'/ELEMENTOR_VERSION', '${PACKAGE_VERSION}'/g" elementor.php
+            PLUGIN_ZIP_FILENAME="elementor-${PACKAGE_VERSION}.zip"
+            npx grunt build
+            mv build elementor
+            zip -r $PLUGIN_ZIP_FILENAME elementor
+            echo "PLUGIN_ZIP_FILENAME=${PLUGIN_ZIP_FILENAME}" >> $GITHUB_ENV
+            echo "PLUGIN_ZIP_FILENAME=${PLUGIN_ZIP_FILENAME}"
+      shell: bash
```
Comment (matipojo) [inline .github/workflows/build-plugin/action.yml:21]: Move before the run

File: .github/workflows/bump-version/action.yml [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    - run: |
+            npm install --no-package-lock --no-save semver@7.3.4
+            PREV_PACKAGE_VERSION=$(node -p "require('./package.json').last_cloud_version")
+            NEW_PACKAGE_VERSION=$(node ./.github/scripts/update-prerelease-version.js ${{ inputs.CHANNEL }})
+            PACKAGE_VERSION=$(node -p "require('./package.json').last_cloud_version")
+            echo "PACKAGE_VERSION=${NEW_PACKAGE_VERSION}" >> $GITHUB_ENV
+            echo "::set-output name=prev_version::${PREV_PACKAGE_VERSION}"
+            echo "PACKAGE_VERSION : ${PACKAGE_VERSION}"
+      shell: bash
+      id: bump_version_step
```
Comment (matipojo) [inline .github/workflows/bump-version/action.yml:25]: Move before the run, id first

File: .github/workflows/config.json [hunk 13]
```diff
@@ -0,0 +1,7 @@
+{
+  "deployment" : { 
+    "envirement" : "dev", 
+    "permitted" : "ronkelementor,KingYes,shilo-ey,matipojo,",
+    "repository" : "elementor"
```
Comment (matipojo) [inline .github/workflows/config.json:5]: Move before the `envirement`

File: .github/workflows/create-tag/action.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (matipojo) [reply .github/workflows/create-tag/action.yml:2]: ?

File: .github/workflows/create-tag/action.yml [hunk 14]
```diff
@@ -0,0 +1,21 @@
+name: Create Git Tag
```
Comment (matipojo) [inline .github/workflows/create-tag/action.yml:1]: Reflect it in action file name

File: .github/workflows/create-tag/action.yml [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+runs:
+  using: "composite"
+  steps:
+    - run: |
+            git config user.name ${{ inputs.MAINTAIN_USERNAME }}
+            git config user.email ${{ inputs.MAINTAIN_EMAIL }}
+            echo "Create tag v${PACKAGE_VERSION}"
+            git tag "v${PACKAGE_VERSION}"
+            git push origin "v${PACKAGE_VERSION}"
+      shell: bash
```
Comment (matipojo) [inline .github/workflows/create-tag/action.yml:21]: Move before the run

File: .github/workflows/generate-changelog/action.yml [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  BASE_TAG_NAME:
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - run: |
+            npm install --no-package-lock --no-save @octokit/core@3.4.0
+            node ./.github/scripts/generate-changelog.js
+      shell: bash
```
Comment (matipojo) [inline .github/workflows/generate-changelog/action.yml:20]: Move before `run`

File: .github/workflows/publish-cloud.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (matipojo) [reply .github/workflows/publish-cloud.yml:1]: Please ensure the file name and action name are the same

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Please target to `master`

File: .github/workflows/build-plugin/action.yml (hunk 11 above)
Comment (maork-elementor) [reply .github/workflows/build-plugin/action.yml:21]: If the "shell: bash" will be before the "run" the action will not work.
As the docs also mentioned, I'm getting also warning on my IDE.

https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

File: .github/workflows/config.json (hunk 13 above)
Comment (maork-elementor) [reply .github/workflows/config.json:5]: done

File: .github/workflows/create-tag/action.yml (hunk 14 above)
Comment (maork-elementor) [reply .github/workflows/create-tag/action.yml:1]: done

File: .github/workflows/create-tag/action.yml (hunk 15 above)
Comment (maork-elementor) [reply .github/workflows/create-tag/action.yml:21]: If the "shell: bash" will be before the "run" the action will not work.
As the docs also mentioned, I'm getting also warning on my IDE.

https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

File: .github/workflows/generate-changelog/action.yml (hunk 16 above)
Comment (maork-elementor) [reply .github/workflows/generate-changelog/action.yml:20]: If the "shell: bash" will be before the "run" the action will not work.
As the docs also mentioned, I'm getting also warning on my IDE.

https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

File: .github/workflows/bump-version/action.yml [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+      run: |
+            npm install --no-package-lock --no-save semver@7.3.4
+            PREV_PACKAGE_VERSION=$(node -p "require('./package.json').last_cloud_version")
+            NEW_PACKAGE_VERSION=$(node ./.github/scripts/update-prerelease-version.js ${{ inputs.CHANNEL }})
+            PACKAGE_VERSION=$(node -p "require('./package.json').last_cloud_version")
+            echo "PACKAGE_VERSION=${NEW_PACKAGE_VERSION}" >> $GITHUB_ENV
+            echo "::set-output name=prev_version::${PREV_PACKAGE_VERSION}"
+            echo "PACKAGE_VERSION : ${PACKAGE_VERSION}"
+      
+
```
Comment (matipojo) [inline .github/workflows/bump-version/action.yml:27]:
```suggestion

```

File: .github/workflows/config.json [hunk 18]
```diff
@@ -0,0 +1,7 @@
+{
+  "deployment" : { 
+    "repository" : "elementor",
+    "envirement" : "dev",
+    "permitted" : "ronkelementor,KingYes,shilo-ey,matipojo,"
```
Comment (matipojo) [inline .github/workflows/config.json:5]:
```suggestion
    "envirement" : "prod",
    "permitted" : "KingYes,shilo-ey,matipojo,"
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Pull request merged**